geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -415,6 +413,14 @@ public long sadd(List<byte[]> membersToAdd, Region<RedisKey, RedisData> region,
     return membersAdded;
   }
 
+  void addForSmove(byte[] memberToAdd, Region<RedisKey, RedisData> region, RedisKey key) {
```

Comment:
I'm concerned that the transactional aspect of the `SMOVE` command will creep in here in a non-obvious way. I would consider adding a check at the lowest possible level to see if we're in a tx and then simply do the `region.put` there. Even so far as just adding it to `AbstractRedisData.storeChanges` although then we'd be creating a bunch of unused delta constructs during tx operations. Alternatively, do this in `sadd` and `srem`. (A tx check can be done by calling `exists()` on the `CachTransactionManager` attached to the `RegionProvider`).

Ideally any redis data structure, (that extends from `AbstractRedisData`), should not contain any tx-specific code.

Reply:
Since moving the transaction check into `AbstractRedisData` would require refactoring all methods that currently call `storeChanges()` to pass in a `RegionProvider` instead of a `Region`, I've moved the check to `sadd` and `srem` for now to limit the impact of this PR.